src/distilabel/llm/vertexai.py

```diff
@@ -0,0 +1,173 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, List, Optional
+
+from pydantic import PrivateAttr
+
+from distilabel.llm.base import AsyncLLM
+from distilabel.utils.logging import get_logger
+
+if TYPE_CHECKING:
+    from vertexai.generative_models import (
+        Content,
+        GenerationConfig,
+        GenerativeModel,
+        Part,
+        SafetySettingsType,
+        Tool,
+    )
+
+    from distilabel.llm.typing import GenerateOutput
+    from distilabel.steps.task.typing import ChatType
+
+
+def _is_gemini_model(model: str) -> bool:
+    """Returns `True` if the model is a model from the Vertex AI Gemini API.
+
+    Args:
+        model (str): the model name to be checked.
+
+    Returns:
+        bool: `True` if the model is a model from the Vertex AI Gemini API.
+    """
+    return "gemini" in model
+
+
+class VertexAILLM(AsyncLLM):
+    """VertexAI LLM implementation running the async API clients for Gemini.
+
+    - Gemini API: https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini
+
+    To use the `VertexAILLM` is necessary to have configured the Google Cloud authentication
+    using one of these methods:
+
+    - Setting `GOOGLE_CLOUD_CREDENTIALS` environment variable
+    - Using `gcloud auth application-default login` command
+    - Using `vertexai.init` function from the `google-cloud-aiplatform` library
+
+    Attributes:
+        model: the model name to use for the LLM e.g. "gemini-pro". [Supported models](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models).
+    """
+
+    model: str
+    _aclient: Optional["GenerativeModel"] = PrivateAttr(...)
+    _content_class: Optional["Content"] = PrivateAttr(...)
+    _part_class: Optional["Part"] = PrivateAttr(...)
+    _generation_config_class: Optional["GenerationConfig"] = PrivateAttr(...)
+    _logger = PrivateAttr(default=get_logger("llm"))
+
+    def load(self) -> None:
+        """Loads the `GenerativeModel` class which has access to `generate_content_async` to benefit from async requests."""
+
+        try:
+            from vertexai.generative_models import (
+                Content,
+                GenerationConfig,
+                GenerativeModel,
+                Part,
+            )
+
+            self._content_class = Content
+            self._part_class = Part
+            self._generation_config_class = GenerationConfig
+        except ImportError as e:
+            raise ImportError(
+                "vertexai is not installed. Please install it using"
+                " `pip install google-cloud-aiplatform`."
+            ) from e
+        except Exception as e:
+            raise e
```

Comment:
is this one needed?

Reply:
Nope, I though I had removed this already.